spec/realtime/message.test.js

```diff
@@ -70,6 +70,50 @@ define(['ably', 'shared_helper'], function(Ably, helper) {
 		}
 	};
 
+	/*
+	 * Test that a message is not sent back to the same realtime client
+	 * when echoMessages is false (RTC1a and RTL7f)
+	 *
+	 * Test that a message is sent back to the same realtime client
+	 * when echoMessages is true (RTC1a and RTL7f)
+	 */
+	exports.publishEcho = function(test) {
+		test.expect(3);
+
+		// set up two realtimes
+		var rt1 = helper.AblyRealtime({ echoMessages: false }),
+			rt2 = helper.AblyRealtime({ echoMessages: true }),
+
+			rt1Channel = rt1.channels.get('publishecho'),
+			rt2Channel = rt2.channels.get('publishecho'),
+
+			testMsg1 = 'Hello',
+			testMsg2 = 'World!';
+
+		monitorConnection(test, rt1);
+		monitorConnection(test, rt2);
+
+		// We expect to see testMsg2 on rt1 and testMsg1 on both rt1 and rt2
+		var expectedMessages = 3;
+
+		rt1Channel.subscribe('event0', function(msg) {
+			test.equal(msg.data, testMsg2, 'Received testMsg2 via rt1 instance');
+			if (!--expectedMessages)
```

Comment:
I personally dislike checking the value of variables you are modifying in place with increment / decrement operators, from the Good Parts of Javascript:

> ++ and --
> The ++ (increment) and -- (decrement) operators have been known to contribute to bad code by encouraging excessive trickiness. They are second only to faulty architecture in enabling to viruses and other security menaces. There is a plusplus option that prohibits the use of these operators.****

It's also really tricky to know if the value is changed before or after evaluation so leads to unreadable code.  Why not put this check into a callback as it's shared by both subscribe callbacks anyway and simplify by first decrementing the value and then checking if all messages are received?